File: res/nashorn2.js
var nashorn = Java.type("org.thushear.tut.nashorn.Nashorn2");

var result = nashorn.fun1("thushear kong");
print("\njs" + result);

nashorn.fun2(123);
nashorn.fun2(49.99);
nashorn.fun2(true);
nashorn.fun2("hi there");
nashorn.fun2(new Date());
nashorn.fun2(new RegExp());
nashorn.fun2({foo:"bar"});

print('passing object hash:');

nashorn.fun3({
    foo:'bar',
    bar:'foo'
});


print('passing custom person object:');

function Person(firstName,lastName){
    this.firstName = firstName;
    this.lastName = lastName;
    this.getFullName = function(){
        return this.firstName + "  " + this.lastName;
    }
}

var person = new Person("thushear","kong");
nashorn.fun3(person);
nashorn.fun4(person);

File: src/main/java/org/thushear/tut/nashorn/Nashorn2.java
package org.thushear.tut.nashorn;

import jdk.nashorn.api.scripting.ScriptObjectMirror;

import javax.script.ScriptEngine;
import javax.script.ScriptEngineManager;
import java.io.FileReader;
import java.util.Arrays;

/**
 * Created by kongming on 2016/3/14
 * Calling java methods from javascript with nashorn.
 */
public class Nashorn2 {


    public static String fun1(String name){
        System.out.format("hello from %s",name);
        return "greeting from java";
    }


    public static String fun2(Object o){
        System.out.println(o.getClass());
        return "type :" + o.getClass();
    }

    public static void fun3(ScriptObjectMirror mirror){
        System.out.println(mirror.getClassName() + ":" + Arrays.toString(mirror.getOwnKeys(true)));
    }


    public static void fun4(ScriptObjectMirror person){
        System.out.println("Full name is : " + person.callMember("getFullName"));
    }



    public static void main(String[] args) throws Exception {

        ScriptEngine engine = new ScriptEngineManager().getEngineByName("nashorn");

        engine.eval(new FileReader("res/nashorn2.js"));

    }

}
